binary_search_tree/tree.py

```diff
@@ -3,57 +3,183 @@ def __init__(self, key, val = None):
         if val == None:
             val = key
 
-        self.key = key
         self.value = val
+        self.key = key
         self.left = None
         self.right = None
-        
-
-
+        # self.val = key
+    
 class Tree:
-    def __init__(self):
+    def __init__(self, height = None):
         self.root = None
+        self.nodes = []
+        #hold for the bfs
+        self.queque = []
+        # self.treeheight = 0
+        self.bfs_nodes = []
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n) because looping through all the nodes
+    # Space Complexity: O(n) because you need to add a node for each key, value
     def add(self, key, value = None):
```

Comment:
👍 The time complexity is O(n log n ) if the tree is balanced and O(n) otherwise.